Formula/mesheryctl.rb

```diff
@@ -0,0 +1,32 @@
+class Mesheryctl < Formula
+  desc "Command-line utility for Meshery, the cloud native management plane"
+  homepage "https://meshery.io"
+  url "https://github.com/meshery/meshery.git",
+      tag:      "v0.6.0-rc.6b",
+      revision: "bb80f4a97c8020aae6624c672ddd2a07e84c2733"
+  version "0.6.0-rc.6b"
+  license "Apache-2.0"
+  head "https://github.com/meshery/meshery.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
+
+    ldflags = %W[
+      -s -w
+      -X github.com/layer5io/meshery/mesheryctl/internal/cli/root/constants.version=v#{version}
+      -X github.com/layer5io/meshery/mesheryctl/internal/cli/root/constants.commitsha=#{Utils.git_short_head}
+      -X github.com/layer5io/meshery/mesheryctl/internal/cli/root/constants.releasechannel=stable
+    ]
+
+    system "go", "build", *std_go_args(ldflags: ldflags), "./mesheryctl/cmd/mesheryctl"
+  end
+
+  test do
+    # We need 'yes' here because mesheryctl needs a meshconfig file to display version
+    # of the server along with the cli. Since it doesn't exist yet it shows a prompt if the user
+    # wants to create a default config file.
+    assert_match version.to_s, shell_output("yes | #{bin}/mesheryctl version 2>&1")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I don't think testing the core functionality of the cli would be possible because almost everything requires a kubernetes cluster to be running. 

However, added a more test case testing the value of release channel. Also, some tests that verify that some main commands such as `mehseryctl system` exist.

Reply:
Done, added one more test.